python/test/unit/language/test_core.py

```diff
@@ -195,7 +196,12 @@ def is_layout_applicable(layout) -> bool:
     if layout in common_layouts:
         return True
     elif is_cuda():
-        return isinstance(layout, MmaLayout)
+        mma_layout = layout.parent if isinstance(layout, DotOperandLayout) else layout
+        if not isinstance(mma_layout, MmaLayout):
+            return False
+        if mma_layout.version[0] >= 3 and not is_hopper():
```

Comment:
What does this condition mean? More specifically, why cannot we test convert layout for `mma_layout.version[0] >= 3` on ampere?

Reply:
so, the way the code that enables `stmatrix` works is by checking the layout version, rather than the capability, so if we test this on Ampere it will generate a `stmatrix` instruction and yell.
I have a patch that looks at the capability rather than the `version` of the `mma` in the `canUseStmatrix` function if you prefer.

Reply:
I just realised I forgot to remove the `is_hopper` part here. I'll add it to my next PR.